fastplotlib/graphics/features/_mesh.py

```diff
@@ -0,0 +1,297 @@
+from typing import Any, Sequence
+
+import numpy as np
+import pygfx
+
+from ._base import (
+    GraphicFeature,
+    GraphicFeatureEvent,
+    to_gpu_supported_dtype,
+    block_reentrance,
+)
+
+from ._positions import VertexPositions
+from ...utils.functions import get_cmap
+from ...utils.triangulation import triangulate
+
+
+def resolve_cmap_mesh(cmap) -> pygfx.TextureMap | None:
+    """Turn a user-provided in a pygfx.TextureMap, supporting 1D, 2D and 3D data."""
+
+    if cmap is None:
+        pygfx_cmap = None
+    elif isinstance(cmap, pygfx.TextureMap):
+        pygfx_cmap = cmap
+    elif isinstance(cmap, pygfx.Texture):
+        pygfx_cmap = pygfx.TextureMap(cmap)
+    elif isinstance(cmap, (str, dict)):
+        pygfx_cmap = pygfx.cm.create_colormap(get_cmap(cmap))
+    else:
+        map = np.asarray(cmap)
+        if map.ndim == 2:  # 1D plus color
+            pygfx_cmap = pygfx.cm.create_colormap(cmap)
+        else:
+            tex = pygfx.Texture(map, dim=map.ndim - 1)
+            pygfx_cmap = pygfx.TextureMap(tex)
+
+    return pygfx_cmap
+
+
+class MeshVertexPositions(VertexPositions):
+    """Manages mesh vertex positions, same as VertexPosition but data must be of shape [n, 3]"""
```

Comment:
This extra class feels like unnecessary bloat to me. I propose to remove it. The `VertexPositions` class already converts 2D data to 3D, so users can create in-plane meshes if they want.